File: app/src/main/java/jgappsandgames/smartreminderslite/sort/tags/UnselectedAdapter.java
package jgappsandgames.smartreminderslite.sort.tags;

// Java
import java.util.ArrayList;

// Views
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

// App
import jgappsandgames.smartreminderslite.R;
import jgappsandgames.smartreminderslite.holder.TagHolder;

// Program
import jgappsandgames.smartreminderssave.tags.TagManager;

/**
 * UnselectedAdapter
 * Created by joshua on 9/2/17.
 */
class UnselectedAdapter extends BaseAdapter {
    private final TagActivity activity;
    private final ArrayList<String> tags;

    UnselectedAdapter(TagActivity activity, ArrayList<String> selected) {
        super();

        this.activity = activity;

        tags = new ArrayList<>();
        for (int i = 0; i < TagManager.tags.size(); i++)
            if (!selected.contains(TagManager.tags.get(i))) tags.add(TagManager.tags.get(i));
    }

    // List Methods
    @Override
    public int getCount() {
        return tags.size();
    }

    @Override
    public int getViewTypeCount() {
        return 1;
    }

    // Item Methods
    @Override
    public String getItem(int position) {
        return tags.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View view, ViewGroup parent) {
        view = LayoutInflater.from(activity).inflate(R.layout.list_tag, parent, false);

        TagHolder holder = new TagHolder(getItem(position), false, activity, view);
        view.setTag(holder);

        return view;
    }
}

File: app/src/main/java/jgappsandgames/smartreminderslite/sort/tags/UnselectedAdapter.kt
package jgappsandgames.smartreminderslite.sort.tags

// Java
import java.util.ArrayList

// Views
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter

// App
import jgappsandgames.smartreminderslite.R
import jgappsandgames.smartreminderslite.holder.TagHolder

// Save
import jgappsandgames.smartreminderssave.tags.TagManager

/**
 * UnselectedAdapter
 * Created by joshua on 1/19/2018.
 */
class UnselectedAdapter(private val activity: TagActivity, selected: ArrayList<String>) : BaseAdapter() {
    private val tags: ArrayList<String> = ArrayList()

    init {
        for (i in TagManager.tags.indices)
            if (!selected.contains(TagManager.tags[i])) tags.add(TagManager.tags[i])
    }

    // List Methods
    override fun getCount(): Int {
        return tags.size
    }

    override fun getViewTypeCount(): Int {
        return 1
    }

    // Item Methods
    override fun getItem(position: Int): String {
        return tags[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convert_view: View, parent: ViewGroup): View {
        var view = convert_view
        view = LayoutInflater.from(activity).inflate(R.layout.list_tag, parent, false)

        val holder = TagHolder(getItem(position), false, activity, view)
        view.tag = holder

        return view
    }
}